exporter/debugexporter/factory.go

```diff
@@ -38,6 +40,14 @@ func NewFactory() exporter.Factory {
 	)
 }
 
+func NewFactoryProfiles() exporterprofiles.Factory {
```

Comment:
We should not put this contract in public.

/cc @mx-psi @dmathieu

Reply:
Is this method even necessary? In the OTLP exporter, we enable profiles within `NewFactory`.

https://github.com/open-telemetry/opentelemetry-collector/blob/403c782d50c630ad945feebe8b26e1f656ccd3c1/exporter/otlpexporter/factory.go#L23-L33

Reply:
Agreed, we can do the same as in the OTLP receiver